smart-contract-extensions/test/unlock-examples/hooks/codeRequiredHook.js

```diff
@@ -0,0 +1,110 @@
+const { constants, protocols } = require('hardlydifficult-ethereum-contracts')
+const erc1820 = require('erc1820')
+const { reverts } = require('truffle-assertions')
+
+const CodeRequiredHook = artifacts.require('CodeRequiredHook.sol')
+
+contract('CodeRequiredHook', accounts => {
+  const lockOwner = accounts[1]
+  const keyBuyer = accounts[5]
+  let lock
+  let hookContract
+  // Never expose the answer or its PK directly in JS for any integrations
+  // Only the answer account's address can safely be exposed
+  let answerAccount
+
+  beforeEach(async () => {
+    // Ganache only setup (a no-op if erc1820 already exists)
+    await erc1820.deploy(web3)
```

Comment:
Maybe use the oz-testhelper here? 
https://docs.openzeppelin.com/test-helpers/0.5/api#erc1820

Reply:
Cool - didn't know that was a thing.

But it's a different use case.  We only use interfaceIds in Solidity for this example... I never use them in JS here.

The line you tagged is about deploying the erc1820 contract to ganache

Reply:
oops, sorry. meant to link to this:
https://docs.openzeppelin.com/test-helpers/0.5/api#erc1820registry

Reply:
OZ is great and all but I'm going to leave this as is for the moment.  It's the same feature, but the library I'm using is the official example from the EIP.

If/when we make this change we can do smart-contracts and smart-contract-extensions at the same time.

Reply:
Oh! I just remembered this step is optional cause `createTestLock` will take care of it.  I'll delete to simplify the test.